.devcontainer/Dockerfile

```diff
@@ -1,30 +1,68 @@
-FROM mcr.microsoft.com/devcontainers/cpp:1-debian-12
+FROM mcr.microsoft.com/devcontainers/cpp:debian-12
 
-# Upgrade the image
-RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
-    && apt-get -y upgrade
+# Set default compiler to clang++
+RUN update-alternatives --install /usr/bin/c++ c++ /usr/bin/clang++ 100
 
-# Install dependencies for static analysis
+# Install tools
 RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
 	&& apt-get -y install --no-install-recommends \
+	# Code formatting
 		clang-format \
+	# Code analysis
 		clang-tidy \
 		cppcheck \
+	# Commenting
 		doxygen \
 		graphviz \
+	# Google Test
 		libgtest-dev \
 		libgmock-dev \
+	# Language Server
 		clangd \
-		bear
+	# Create compile_commands.json
+		bear \
+	# Networking tools
+		netcat-openbsd \
+		telnet \
+		siege \
+	# Dependencies for kcov
+		binutils-dev \
+		libssl-dev \
+		libcurl4-openssl-dev \
+		libelf-dev \
+		libstdc++-12-dev \
+		zlib1g-dev \
+		libdw-dev \
+		libiberty-dev \
+	# Remove valgrind because we build newer version
+	&& apt-get remove -y \
+		valgrind \
+	&& rm -rf /var/lib/apt/lists/*
 
-# Setup googletest
-RUN mkdir -p $HOME/build \
-	&& cd $HOME/build \
+# Build googletest, kcov, and valgrind
```

Comment:
Would separate these cmds for 
- googletest
- kcov
- valgrind
Provides clearer structure and also permits execution of individual packages in case a previous cmd fails for some reason.

Reply:
Did you mean separate into three RUN commands?
The reason behind combining was to reduce Docker layers: I read that you should reduce layers to keep the image size small. 

But it makes no difference here: image size stayed at 2.48gb, independent if one or three RUN commands. 
So splitting makes sense since we can cash the different steps. 
Made three RUN commands

I moved the valgrind building up, since it takes the longest to build.

I also removed apt install for gtest/gmock and use the same style of cmake, make, make install > reduced image size to 2.47gb

>  in case a previous cmd fails

If you meant separate the three commands inside the RUN command: I don't think we want to continue if something fails. The we would have a half init container

Reply:
I meant three RUN cmds. This way we would still execute the other RUN cmds, if one of them fails or am I wrong in this assumption? I think it is in our interest to have the container continue the build process as we could still work with it if not all tools are set up. Many of them are optional like the coverage tool or we can still add them afterwards.

Reply:
But closing the conversation as the requested changes are introduced.